packages/api/src/base/Decorate.ts

```diff
@@ -454,7 +453,10 @@ export abstract class Decorate<ApiType extends ApiTypes> extends Events {
       ...Array(Math.ceil(keys.length / PAGE_SIZE_VALS))
         .fill(0)
         .map((_, index): Observable<Codec[]> =>
-          this._rpcCore.state.subscribeStorage(keys.slice(index * PAGE_SIZE_VALS, (index * PAGE_SIZE_VALS) + PAGE_SIZE_VALS))
+          ((this.hasSubscriptions || !this._rpcCore.state.queryStorageAt)
```

Comment:
There is a slight issue here - if the provider connects to a chain without subscriptions and no queryStorageAt, it will still try to use the subscription. (This is unlike the original where this was just not available)

The `_decorateMulti ` above has the same issue.

Reply:
Now the decoration will be skipped if `this.supportMulti` is false.

Reply:
Thank you, will take  proper look in the next short while - looks promising from a cursory glance.